packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -4982,15 +4982,15 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   /// ```
   /// {@end-tool}
   @optionalTypeArgs
-  void pop<T extends Object?>([ T? result ]) {
+  Future<void> pop<T extends Object?>([ T? result ]) async {
```

Comment:
I wouldn't worry too much about the lint. We don't recommend people using it because it has other problems with flutter APIs. It's better to be truthful about this being potentially async.

Reply:
agree, in that case, please also update the Navigator.pop and popuntil